exec_helpers/subprocess_runner.py

```diff
@@ -268,12 +270,19 @@ def poll_pipes(
             # Run
             self.__process = subprocess.Popen(
                 args=[command],
-                stdin=subprocess.PIPE,
                 stdout=subprocess.PIPE if open_stdout else devnull,
                 stderr=subprocess.PIPE if open_stderr else devnull,
-                shell=True, cwd=cwd, env=env,
+                stdin=subprocess.PIPE,
+                shell=True,
+                cwd=cwd,
+                env=env,
                 universal_newlines=False,
             )
+            if stdin is not None:
+                if not isinstance(stdin, six.text_type):
+                    stdin = six.u(stdin)
```

Comment:
better to decode for `ExecResult` inside `ExecResult` and encode for stdin only if required (less transformations)